language/vm/vm-runtime/src/libra_vm.rs

```diff
@@ -415,11 +415,13 @@ impl LibraVM {
                         self.execute_user_transactions(txns, &mut data_cache, state_view)?;
                     result.append(&mut outs);
                 }
-                TransactionBlock::BlockPrologue(block_metadata) => {
-                    result.push(self.move_vm.execute_runtime(|runtime| {
-                        process_block_metadata(block_metadata, runtime, &mut data_cache)
-                    })?)
-                }
+                TransactionBlock::BlockPrologue(block_metadata) => result.push(
+                    self.move_vm
+                        .execute_runtime(|runtime| {
+                            process_block_metadata(block_metadata, runtime, &mut data_cache)
+                        })
+                        .unwrap_or_else(discard_error_output),
```

Comment:
Was this a bug? Does this change the behavior of the prologue?

Reply:
This will only change the behavior when BlockPrologue triggers an error, which is unexpected anyway. The success case will remain the same

Reply:
Does one of the tests exercise this new behavior?

Reply:
Yes. The failed block prologue transaction will not propagate error to the caller but rather generate a TransactionOutput with a Discarded status.

Reply:
curious why you made this change, isnt' the implicit `?` beautiful(I wrote it lol)?  the semantic doesn't change since `TransactionOutput impl From<VMStatus>`